Design.md

```diff
@@ -227,8 +229,26 @@ This API will enable or disable user on Vault.
 
 
 ### Create S3 Credential
-This API creates S3 credential for the user.
-1. `generate-access-key` api will be called using assumed role credentials.
+This API creates S3 credentials for the user.
+1. The Generate Access Key API will be called using assumed role credentials.
+1. Encrypt the secret key (For more, see [SecretKey-Encryption-Strategy](#SecretKey-Encryption-Strategy)). Store the encrypted secret key on Redis Sentinel in the hash named "osis:s3credentials", with the hash key formatted as `<Username>__<AccessKeyID>`.
+1. Log the access key ID _(Log location and tools for logging are yet to be decided)_
+
+**Important Notes:**
+* Redis Sentinel is subject to an ongoing rolling upgrade.
+* If the entire Redis Sentinel cluster fails or crashes, the vCloud Director's OSE service must be restarted with the following command:
+    ```shell
+    $ ose service restart
+    ```
+  * Once OSE service restart process is finished,
+     * OSE will invoke `listS3Credentials` API to identify the keys for object storage operations. OSIS will return only an empty list for `listS3Credentials` API (as no key  is available for the user on Redis).
+     * If no keys are present for a user, OSE will invoke `createS3Credentials` API to create a new key for object storage operations.
+  * Access keys created by OSIS before OSE restart time (`ose-restart-time`) will not be listed by OSIS. vCloud Director Tenant Users are responsible to clean up unused access keys using S3 console or IAM API because they are the only one who knows if the access keys are in used or not.
+  * Access key IDs generated by OSIS will be logged
+  * Redis Sentinel only supports storing **4294967295** (approx. 4.2 billion) keys in the hash.
```

Comment:
```suggestion
* Redis Sentinel only supports storing **4294967295** (approx. 4.2 billion) keys in the hash.
```

Reply:
fixed in a7b6b598aa22174d036796d548261821f49edc71